pkg/csr/csr.go

```diff
@@ -0,0 +1,236 @@
+package csr
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"net"
+	"reflect"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	certificatesv1 "k8s.io/api/certificates/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/kubernetes"
+)
+
+//+kubebuilder:rbac:groups="certificates.k8s.io",resources=certificatesigningrequests/approval,verbs=update
+//+kubebuilder:rbac:groups="certificates.k8s.io",resources=certificatesigningrequests,verbs=get;list;watch
+//+kubebuilder:rbac:groups="certificates.k8s.io",resources=signers,verbs=Approve,resourceNames=kubernetes.io/kube-apiserver-client-kubelet;kubernetes.io/kubelet-serving
+
+// kubeletClientUsages contains the permitted key usages from a kube-apiserver-client-kubelet signer
+var kubeletClientUsages = []certificatesv1.KeyUsage{
```

Comment:
Looking at how this is being used, wouldn't it be easier if this was a set?

Reply:
this was taken from the way cluster-machine-approver uses it, I have added the link to approver code in the commit msg and the CSR package header now.
https://github.com/openshift/cluster-machine-approver/blob/master/pkg/controller/node_client.go#L32